wgpu-hal/src/metal/adapter.rs

```diff
@@ -395,161 +391,95 @@ impl crate::Adapter<super::Api> for super::Adapter {
 
 const RESOURCE_HEAP_SUPPORT: &[MTLFeatureSet] = &[
     MTLFeatureSet::iOS_GPUFamily1_v3,
-    MTLFeatureSet::iOS_GPUFamily2_v3,
```

Comment:
This looks wrong to me. Is the change based on this quote from their docs?
> More recent GPUs have higher version numbers and support larger feature sets. A higher GPU version is always a superset of an earlier version in the same GPU family.

It says `iOS_GPUFamily1_v3` includes `iOS_GPUFamily1_v2` and `iOS_GPUFamily1_v1`.
It doesn't say that `iOS_GPUFamily2_v3` includes `iOS_GPUFamily1_v3`, which is what the change seems to imply here.

Reply:
> Metal organizes GPUs into four major families:
Use the Common family to create apps that target a range of GPUs on multiple platforms.
Use the Apple family to create apps that target Apple GPUs.
Use the Mac family to create apps that target GPUs on macOS systems.
Use the Mac Catalyst family when building an iPadOS app to run on macOS.

`iOS_GPUFamilyX_vX` all the same Apple family.

Reply:
Interesting, thank you for the links and quotes!